lib/tokens.js

```diff
@@ -0,0 +1,687 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as base64url from 'base64url-universal';
+import * as bedrock from 'bedrock';
+import * as database from 'bedrock-mongodb';
+import {tokenizers} from 'bedrock-tokenizer';
+import {IdGenerator, IdEncoder} from 'bnid';
+import {randomBytes, timingSafeEqual} from 'crypto';
+import pLimit from 'p-limit';
+import {promisify} from 'util';
+import {createKek} from './aeskw.js';
+import Bitstring from '@digitalbazaar/bitstring';
+import * as tokenVersions from './tokenVersions.js';
+const randomBytesAsync = promisify(randomBytes);
+const {util: {BedrockError}} = bedrock;
+
+// initial bitstring for batchIndexSize=1
+const BATCH_INDEX_SIZE_1 = 'H4sIAAAAAAAAA2NgwA8ArVUKGSAAAAA';
+
+const VERSION_SIZE = 2;
+const UNWRAPPED_SIZE = 24;
+const WRAPPED_SIZE = UNWRAPPED_SIZE + 8;
+const MAX_AAD_SIZE = 8;
```

Comment:
Can the constants other than VERSION_SIZE depend on the version value?  If so, then perhaps they should be removed in favor of using dynamic sizes values based on the version.

Reply:
Yes, these will likely be removed in favor of supporting any dynamic size at this level based on the input parameters. Applications using this library can set constraints appropriate to their use case.